### PR TITLE
Added a section on 'btrfs balance' (bsc#1170577)

### PR DESCRIPTION
**Pull request by tbazant**
### Description

A new section describes what `btrfs balance` does and how to customize its behaviour

### Are there any relevant issues/feature requests?

* bsc#1170577

### To which product versions do the changes apply?

The first column is to be filled by the requester, the second column is to be filled by the doc team after the changes have been backported to the maintenance branches.

| Code 15     | Backport Done
| ----------  | ---------- 
| [x] 15 SP3  | (`master` only)   
| [x] 15 SP2  | [ ] 
| [x] 15 SP1  | [ ] 
| [x] 15 SP0  | [ ] 

| Code 12     | Backport Done
| ----------  | ---------- 
| [x] 12 SP5  | [ ] 
| [x] 12 SP4  | [ ] 
| [x] 12 SP3  | [ ] 
| [ ] 12 SP2  | [ ]

### CODE REVIEW

File: xml/storage_filesystems.xml [hunk 1]
```diff
@@ -2017,12 +2017,43 @@ Label: none uuid: 40123456-cb2c-4678-8b3d-d014d1c78c78
...
+     of the <package>btrfsmaintenance</package> package.
     </para>
    </warning>
   </sect2>
 
+  <sect2 xml:id="sec-filesystems-trouble-balance">
+   <title>Btrfs: Balancing Data across Devices</title>
+   <para>
+    The <command>btrfs balance</command> command is part of the
+    <package>btrfsmaintenance</package> package. It balances block groups on a
```
Comment (lproven) [inline xml/storage_filesystems.xml:2029]:
```suggestion
    <package>btrfsmaintenance</package> package. It balances block groups on
```

File: xml/storage_filesystems.xml [hunk 2]
```diff
@@ -2017,12 +2017,43 @@ Label: none uuid: 40123456-cb2c-4678-8b3d-d014d1c78c78
...
+    Btrfs file systems. For example, if you have 1 TB drive with 600 GB used by
+    data and you add another 1 TB drive, the balancing will theoretically
+    result in having 300 GB used space on each drive.
```
Comment (lproven) [inline xml/storage_filesystems.xml:2032]:
```suggestion
    Btrfs file systems. For example, if you have 1&nbsp;TB drive with 600&nbsp;GB used by
    data and you add another 1&nbsp;TB drive, the balancing will theoretically
    result in having 300&nbsp;GB used space on each drive.
```

File: xml/storage_filesystems.xml [hunk 3]
```diff
@@ -2017,12 +2017,43 @@ Label: none uuid: 40123456-cb2c-4678-8b3d-d014d1c78c78
...
+   </para>
+   <para>
+    The balancing can also clear out non-full portions of block devices and
+    spread data more evenly. Similarly, if you have a lot of near-empty chunks,
+    their space will not be available until the balancing has cleared those
+    chunks.
+   </para>
+   <tip>
+    <para>
+     To fine tune the default behavior of balancing data on Btrfs file
```
Comment (lproven) [inline xml/storage_filesystems.xml:2042]:
```suggestion
     To fine-tune the default behavior of balancing data on Btrfs file
```

File: xml/storage_filesystems.xml [hunk 4]
```diff
@@ -2017,12 +2017,43 @@ Label: none uuid: 40123456-cb2c-4678-8b3d-d014d1c78c78
...
+     systems&mdash;for example, how frequently or which mount points to
+     balance&mdash; inspect and customize
+     <filename>/etc/sysconfig/btrfsmaintenance</filename>. The relevant options
+     start with <option>BTRFS_BALANCE_</option>.
+    </para>
+   </tip>
+   <para>
+    Alternatively, you can use the balancing to migrate data between different
+    RAID types. For example, to convert data on a set of disks from RAID1 to
+    RAID5 you can run the following command:
```
Comment (lproven) [inline xml/storage_filesystems.xml:2052]:
```suggestion
    RAID5, you can run the following command:
```

**Review by lproven — CHANGES_REQUESTED** [review]
Only small stuff!

File: xml/storage_filesystems.xml (hunk 2 above)
Comment (goldwynr) [inline xml/storage_filesystems.xml:2032]: This is a good example. A better example will be of usage, that is compacting half empty block groups based on usage. That will justify why it needs to be run weekly.

**Review by goldwynr — CHANGES_REQUESTED** [review]

**Review by lproven — APPROVED** [review]

File: xml/storage_filesystems.xml [hunk 5]
```diff
@@ -2026,10 +2026,10 @@ Label: none uuid: 40123456-cb2c-4678-8b3d-d014d1c78c78
    <title>Btrfs: Balancing Data across Devices</title>
    <para>
     The <command>btrfs balance</command> command is part of the
-    <package>btrfsmaintenance</package> package. It balances block groups on a
-    Btrfs file systems. For example, if you have 1 TB drive with 600 GB used by
-    data and you add another 1 TB drive, the balancing will theoretically
-    result in having 300 GB used space on each drive.
+    <package>btrfsmaintenance</package> package. It balances block groups on
+    Btrfs file systems. For example, if you have 1&nbsp;TB drive with 600&nbsp;GB used by
```
Comment (goldwynr) [inline xml/storage_filesystems.xml]: This still shows btrfs balance a part of btrfsmaintenance. btrfs balance is a part of btrfs-progs.

**Review by goldwynr — COMMENTED** [review]
Please correct the package name. btrfsmaintenance should be replaced by btrfs-progs.

File: xml/storage_filesystems.xml (hunk 5 above)
Comment (tbazant) [reply xml/storage_filesystems.xml]: > This still shows btrfs balance a part of btrfsmaintenance. btrfs balance is a part of btrfs-progs.

This should have already been fixed. Refresh and verify please.

**Pull request merged**